### PR TITLE
*: fix set/get var race

### PR DESCRIPTION
**Pull request by hanfei1991**
We may read and write user variable at same time, so I use a rw lock to avoid data race.
@zimulala @zz-jason @XuHuaiyu PTAL

### CODE REVIEW

File: sessionctx/variable/session.go [hunk 1]
```diff
@@ -100,6 +100,7 @@ func (tc *TransactionContext) UpdateDeltaForTable(tableID int64, delta int64, co
 
 // SessionVars is to handle user-defined or global variables in the current session.
 type SessionVars struct {
+	sync.RWMutex
```
Comment (zimulala) [inline sessionctx/variable/session.go]: It's used for `Users`, so I think to add a mutex for the Users?

Comment (hanfei1991) [inline sessionctx/variable/session.go]: It's more convenient to do this ~

Comment (zimulala) [inline sessionctx/variable/session.go]: But it's confusing.

**Comment by hanfei1991** [comment]
@zimulala @XuHuaiyu PTAL

**Comment by zimulala** [comment]
LGTM

**Comment by XuHuaiyu** [comment]
LGTM

**Pull request merged**